src/Integration/ProjectToLanguageMapper.cs

```diff
@@ -93,11 +100,14 @@ private Language GetLanguageForProject(Project dteProject)
                 return language;
             }
 
-            if (projectKind == OpenAsFolderProject)
+            var isOpenAsFolder = folderWorkspaceService.IsFolderWorkspace();
+
+            if (isOpenAsFolder)
             {
-                var cmakeProjectFile = absoluteFilePathLocator.Locate("CMakeLists.txt");
+                var rootDirectory = folderWorkspaceService.FindRootDirectory();
+                var isCMake = fileSystem.Directory.EnumerateFiles(rootDirectory, "CMakeLists.txt", SearchOption.AllDirectories).Any();
```

Comment:
@duncanp-sonar I decided to go with fileSystem search. I opened sonar-clang multiple times and absoluteFilePathLocator kept returning null, even after waiting for a long time for VS to finish loading. Then once it did work. And then it stopped working again.

All in all, the behavior was inconsistent so I switched to using the file system. We had a similar problem with TsConfigLocator, we started by using IAbsoluteFilePathLocator and switched to file system.
https://github.com/SonarSource/sonarlint-visualstudio/pull/2387

I think we should revisit using IAbsoluteFilePathLocator at all, we use it in hotspots and it's highly inconsistent.